OMCompiler/Compiler/BackEnd/BackendDAECreate.mo

```diff
@@ -1104,7 +1105,7 @@ algorithm
 
     else
       algorithm
```

Comment:
Maybe switch from algorithm to equation for this case. Since we assign to a value that is not a CREF

Reply:
I don't really get why this makes a difference, but if you think its better i will change it for the other PR.

Reply:
I actually discussed this with Mahder today, it is interesting that we in general allow assignment to something that is not a CREF  in MetaModelica (Variable, like in C). I have seen it at places but for a situation like this we might want to use equation instead since we check for equality and are actually not assigning something (Just a minor thing, you do not have to change it)

Reply:
So in general it is better to use equation when using actual equality checks? Seems reasonable. 

But i think best practice would be to use guard() instead ... this is an else case so maybe not here (although i am a bit confused about it).

Reply:
> 
> 
> So in general it is better to use equation when using actual equality checks? Seems reasonable.
> 
> But i think best practice would be to use guard() instead ... this is an else case so maybe not here (although i am a bit confused about it).

I agree with you 100%! It would also be more efficient, I believe